sphinx_immaterial/apidoc/cpp/api_parser.py

```diff
@@ -70,10 +72,14 @@
 import pydantic.dataclasses
 import sphinx.domains.cpp
 import sphinx.util.logging
-from typing_extensions import TypedDict, NotRequired
 
 from . import ast_fixes  # pylint: disable=unused-import
 
+if sys.version_info >= (3, 11):
+    from typing import NotRequired
+else:
+    from typing_extensions import NotRequired
```

Comment:
This logic is basically already present in typing_extensions, but I'm not sure it is worth moving this condition into our code to avoid a dependency on typing_extensions when using 3.11, given that typing_extensions may likely already be installed and is pretty lightweight.  It is likely that we will depend on other stuff from typing_extensions in the future, so we may end up having to add the dependency back even for 3.11.

Reply:
You want me to revert this and re-add the dependency to the theme (instead of the cpp optional deps)?

Reply:
Yeah I think it makes sense to just keep an unconditional dependency on typing_extensions

Reply:
done